ValveResourceFormat/Resource/ResourceTypes/Sound.cs

```diff
@@ -144,13 +169,19 @@ public override void Read(BinaryReader reader, Resource resource)
             SampleCount = reader.ReadUInt32();
             Duration = reader.ReadSingle();
 
-            // Skipping over m_Sentence (CSentence_t) and m_pHeader
-            reader.BaseStream.Position += 12;
+            var sentenceOffset = reader.ReadInt64();
+            
+            if (sentenceOffset != 0)
+                sentenceOffset = reader.BaseStream.Position + sentenceOffset;
+            
+            // Skipping over m_pHeader
+            reader.BaseStream.Position += 4;
 
             StreamingDataSize = reader.ReadUInt32();
 
             if (resource.Version < 1)
```

Comment:
Can you flip the ifs so there's no early returns?

Reply:
Done.